azkaban-common/src/main/java/azkaban/utils/Utils.java

```diff
@@ -325,6 +327,18 @@ public static String formatDuration(long startTime, long endTime) {
     return days + "d " + hours + "h " + minutes + "m";
   }
 
+  public static void setCurrentTimeMillisFixed(long millis) {
```

Comment:
This is only used for unit testing, right?
How about changing the name to something like setCurrentTimeMillisForTesting? 
Or alternatively add a comment. 

Ideally, I would prefer using a mocking framework to mock out the call to System.currentTimeMillis. However the change required would be much more significant. What you have done is an improvement. Having a test is much better than not having one!